src/main/scala/com/exasol/cloudetl/kinesis/KinesisShardDataImporter.scala

```diff
@@ -0,0 +1,87 @@
+package com.exasol.cloudetl.kinesis
+
+import scala.collection.JavaConverters._
+
+import com.exasol.cloudetl.util.JsonDeserializer
+
+import com.amazonaws.auth._
+import com.amazonaws.services.kinesis.{AmazonKinesis, AmazonKinesisClientBuilder}
+import com.amazonaws.services.kinesis.model._
+import com.exasol._
+import java.util
+
+object KinesisShardDataImporter {
+  def run(exaMetadata: ExaMetadata, exaIterator: ExaIterator): Unit = {
+    val kinesisArgumentsReader = new KinesisArgumentsReader(exaIterator)
+    val awsAccessKeyId = kinesisArgumentsReader.getAwsAccessKeyArgument
+    val awsSecretAccessKey = kinesisArgumentsReader.getAwsSecretKeyArgument
+    val awsSessionToken = kinesisArgumentsReader.getAwsSessionTokenArgument
+    val streamName = kinesisArgumentsReader.getStreamNameArgument
+    val region = kinesisArgumentsReader.getRegionArgument
+    val shardId = kinesisArgumentsReader.getShardIdArgument
+    val shardSequenceNumber = kinesisArgumentsReader.getShardSequenceNumber
+    val awsCredentials =
+      new BasicSessionCredentials(awsAccessKeyId, awsSecretAccessKey, awsSessionToken)
+    val amazonKinesis = AmazonKinesisClientBuilder.standard
+      .withRegion(region)
+      .withCredentials(new AWSStaticCredentialsProvider(awsCredentials))
+      .build
+    val shardIteratorRequest =
+      createShardIteratorRequest(streamName, shardId, shardSequenceNumber)
+    val records = getRecords(amazonKinesis, shardIteratorRequest)
+    try {
+      records.foreach(record => {
+        exaIterator.emit(createTableValuesListFromRecord(record, shardId): _*)
+      })
+    } catch {
```

Comment:
Check if you need to close the kinesis client after consuming the records. If so, close it inside the finally clause.

Reply:
added `shutdown`